SwiftSyntaxDevUtils/Sources/swift-syntax-dev-utils/commands/LocalPrPrecheck.swift

```diff
@@ -0,0 +1,51 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import ArgumentParser
+import Foundation
+
+struct LocalPrPrecheck: ParsableCommand {
+  static let configuration = CommandConfiguration(
+    abstract: """
+      Ensure changes are fully tested, formatted, and validated before pull request submission.
+      """
+  )
+
+  @Flag(help: "Enable verbose logging.")
+  var verbose: Bool = false
+
+  func run() throws {
+    let executor = LocalPrPrecheckExecutor(verbose: verbose)
+    try executor.run()
+  }
+}
+
+struct LocalPrPrecheckExecutor {
+  private let formatExecutor: FormatExecutor
+  private let generateSourceCodeExecutor: GenerateSourceCodeExecutor
+  private let buildExecutor: BuildExecutor
+  private let testExecutor: TestExecutor
+
+  init(verbose: Bool = false) {
+    self.formatExecutor = FormatExecutor(update: false)
+    self.generateSourceCodeExecutor = GenerateSourceCodeExecutor(toolchain: nil, verbose: verbose)
+    self.buildExecutor = BuildExecutor(swiftPMBuilder: SwiftPMBuilder(toolchain: nil, useLocalDeps: false))
+    self.testExecutor = TestExecutor(swiftPMBuilder: SwiftPMBuilder(toolchain: nil, useLocalDeps: false))
```

Comment:
Should we also pass down `verbose` to `BuildExecutor` and `TestExecutor`?

Reply:
Good point, thanks. Done.